spock-core/src/main/java/spock/util/time/MutableClock.java

```diff
@@ -0,0 +1,165 @@
+package spock.util.time;
+
+import java.time.*;
+import java.time.temporal.*;
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+/**
+ * A mutable implementation of {@link Clock} for testing purposes.
+ *
+ * @author Leonard Brünings
+ * @since 2.0
+ */
+public class MutableClock extends Clock {
+
+  private final ZoneId zone;
+  private Instant instant;
+
+  /**
+   * {@link TemporalAmount} used for {@link #next()} and {@link #previous()} changes.
+   */
+  private TemporalAmount changeAmount;
+
+  /**
+   * Initializes a with {@link Instant#now()} and {@link ZoneId#systemDefault()} and 1 second {@link #changeAmount}.
+   */
+  public MutableClock() {
+    this(Instant.now(), ZoneId.systemDefault());
+  }
+
+  /**
+   * Initializes a with given instant and {@link ZoneId#systemDefault()} and 1 second {@link #changeAmount}.
+   * @param instant to use
+   */
+  public MutableClock(Instant instant) {
+    this(instant, ZoneId.systemDefault());
+  }
+
+  /**
+   * Initializes a with {@link Instant#now()} and given zone and 1 second {@link #changeAmount}.
+   * @param zone to use
+   */
+  public MutableClock(ZoneId zone) {
+    this(Instant.now(), zone);
+  }
+
+  /**
+   * Initializes a with 1 second {@link #changeAmount} and given parameters.
+   * @param instant to use
+   * @param zone to use
+   */
+  public MutableClock(Instant instant, ZoneId zone) {
+    this(instant, zone, Duration.ofSeconds(1));
+  }
+
+  public MutableClock(Instant instant, ZoneId zone, TemporalAmount changeAmount) {
+    this.instant = Objects.requireNonNull(instant,"instant may not be null");
+    this.zone = Objects.requireNonNull(zone, "zone may not be null");
+    this.changeAmount = Objects.requireNonNull(changeAmount, "changeAmount may not be null");
+  }
+
+  /**
+   * Initializes a with instant and zone from {@link ZonedDateTime} and 1 second {@link #changeAmount}.
+   * @param zonedDateTime to use for instant and zone information
+   */
+  public MutableClock(ZonedDateTime zonedDateTime) {
+    this(zonedDateTime.toInstant(), zonedDateTime.getZone());
+  }
+
+  @Override
+  public ZoneId getZone() {
+    return zone;
+  }
+
+  @Override
+  public MutableClock withZone(ZoneId zone) {
+    return new MutableClock(instant, zone, changeAmount);
+  }
+
+  @Override
+  public Instant instant() {
+    return instant;
+  }
+
+  public void setChangeAmount(TemporalAmount changeAmount) {
+    assert changeAmount != null : "changeAmount may not be null";
```

Comment:
Are you sure you pushed it?

Reply:
the github update-merge blocked my change and I didn't see the dialog in intellij, should be fixed now.